src/Features/LanguageServer/Protocol/Handler/Symbols/RoslynDocumentSymbol.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    /// <summary>
+    /// Derived version of DocumentSymbol used so we can pass additional information we do not want lost for our
+    /// internal consumers.  For example, the richer <see cref="Glyph"/> which is a superset of <see
+    /// cref="DocumentSymbol.Kind"/>.
+    /// </summary>
+    internal sealed class RoslynDocumentSymbol : DocumentSymbol
+    {
+        [DataMember(IsRequired = false, Name = "glyph")]
+        public int Glyph { get; set; }
+
+        // Deliberately override the value in the base.
+        [DataMember(Name = "children")]
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public new RoslynDocumentSymbol[]? Children { get; set; }
```

Comment:
> Can/should this be forwarding to the base property rather than just defining a new one?

Let me try that!  

> is this so we have a place to hold our custom types once we've deserialized the JSON and reparented everything? 

Yes.  If we don't do this, then the serializer tries to serialize/deserialize 'children' without the extra data (like the 'glyph' field).

Reply:
changed this to use the same underlying storage field though.